flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/asyncprocessing/AbstractAsyncStateStreamOperator.java

```diff
@@ -259,6 +262,44 @@ public Object getCurrentKey() {
         return currentProcessingContext.getKey();
     }
 
+    @Override
+    public void processWatermark(Watermark mark) throws Exception {
+        if (!isAsyncStateProcessingEnabled()) {
+            // If async state processing is disabled, fallback to the super class.
+            super.processWatermark(mark);
+            return;
+        }
+        asyncExecutionController
+                .getEpochManager()
+                .onNonRecord(
+                        () -> {
+                            try {
+                                if (timeServiceManager != null) {
+                                    timeServiceManager.advanceWatermark(mark);
+                                }
+                                output.emitWatermark(mark);
+                            } catch (Exception e) {
+                                handleAsyncStateException(
+                                        String.format("Failed to process watermark %s.", mark), e);
+                            }
+                        },
+                        ParallelMode.SERIAL_BETWEEN_EPOCH);
```

Comment:
My suggestion:
1. `ParallelMode` could be configured within `EpochManager` or `AEC`, and could be overridden from outside.
2. `AEC` should proxy the requests to `EpochManager`, the operator does not know the details.